raiden/raiden_event_handler.py

```diff
@@ -195,30 +195,24 @@ def handle_paymentsentsuccess(
             payment_sent_success_event: EventPaymentSentSuccess,
     ):
         target = payment_sent_success_event.target
-        payment_status = raiden.pop_payment_status(
-            target=target,
-            payment_identifier=payment_sent_success_event.identifier,
-        )
+        payment_identifier = payment_sent_success_event.identifier
+        payment_status = raiden.targets_to_identifiers_to_statuses[target].pop(payment_identifier)
 
-        if payment_status:
-            payment_status.payment_done.set(True)
-        else:
-            # can happen: See
-            # https://github.com/raiden-network/raiden/issues/3121#issuecomment-448664092
-            log.warning(
-                'Payment success without payment status',
-                event=payment_sent_success_event,
-            )
+        # With the introduction of the lock we should never get an async result
+        # always get here only once per identifier so payment_status should always exist
+        # see: https://github.com/raiden-network/raiden/pull/3191
+        payment_status.payment_done.set(True)
 
     def handle_paymentsentfailed(
             self,
             raiden: RaidenService,
             payment_sent_failed_event: EventPaymentSentFailed,
     ):
         target = payment_sent_failed_event.target
-        payment_status = raiden.pop_payment_status(
-            target=target,
-            payment_identifier=payment_sent_failed_event.identifier,
+        payment_identifier = payment_sent_failed_event.identifier
+        payment_status = raiden.targets_to_identifiers_to_statuses[target].pop(
+            payment_identifier,
+            None,
         )
         # In the case of a refund transfer the payment fails earlier
         # but the lock expiration will generate a second
```

Comment:
Something is wrong here, there should only be one event.

A refund transfer to the /initiator/ means the *transfer* failed, but it does not necessarily means the payment failed.